### DISCUSSION TITLE
Is there Documentation with a List of Allowed Values for Events?

### DISCUSSION THREAD

**Discussion by jlev23** [weather-gov/api#644] [Q&A] [answered]
Is there a list of allowed values for events? i.e.`"event": "Small Craft Advisory",`

**Comment by netbymatt** [comment, marked as answer]
I believe you're looking for this list
https://api.weather.gov/alerts/types

**Reply by jlev23** [reply]
Thanks @netbymatt! Do you know if the alert types have a corresponding code? i.e. "SCY" for "small craft advisory"

**Reply by netbymatt** [reply]
The /products endpoint has some things listed but it doesn't seem to be a 1:1 relationship to /alerts
https://api.weather.gov/products/types

**Reply by scadergit** [reply]
It's a little more complicated, but effectively it's "a concatenation of the two character Phenomena (PP) and one character Significance (S) code from the NWS Valid Time Event Code (VTEC)" (quoted from the [NWS CAP 1.2 specification](https://www.weather.gov/media/alert/CAP_v12_guide_05-16-2017.pdf), search for "eventCode"). There are additional links within that which provide common examples. These codes can be used in the API alerts endpoint using the `code` parameter which accepts a common-separated list of codes.

**Reply by scadergit** [reply]
I happened to be discussing alerts with the offices that manage NWS CAP products and they provided this list of alert event codes:
https://vlab.noaa.gov/web/nws-common-alerting-protocol/cap-documentation#eventcode

**Reply by dsteketee76** [reply]
eventCode doesn't appear in the Alerts response object (https://api.weather.gov/alerts/active) as far as I've seen.

**Comment by JJKraw** [comment]
There is also information at https://www.weather.gov/help-map but you have to scrape HTML to get it. The page is a key to the map at the weather.gov home page. The alerts are sorted into priority order (I use this info to decide the order to display multiple alerts for a location) and the colors used on the map (note that "transparent" is spelled incorrectly ;) ).

I know the NWS crew doesn't need any more work but it would be nice to toss this onto the API wish list.
